main/bluetooth/hidp/ps.c

```diff
@@ -45,6 +47,74 @@ static void bt_hid_cmd_ps5_rumble_init(struct bt_dev *device) {
     bt_hid_cmd_ps5_set_conf(device, (void *)&ps5_set_conf);
 }
 
+static void bt_hid_cmd_ps5_trigger_init(struct bt_dev *device) {
+    int32_t perc_threshold_l = -1;
+    int32_t perc_threshold_r = -1;
+    int32_t dev = bt_host_get_dev_from_out_idx(device->ids.out_idx, &device);
+    uint32_t map_cnt_l = 0;
+    uint32_t map_cnt_r = 0;
+
+    /* Go through the list of mappings, looking for PAD_RM and PAD_LM */
+    for (uint32_t i = 0; i < config.in_cfg->map_size; i++) {
+        /* Threshold doesn't make sense when the destination is axis, skip */
+        if (btn_is_axis(config.in_cfg[dev].map_cfg[i].dst_id, config.in_cfg[dev].map_cfg[i].dst_btn)) {
+            continue;
+        }
+        /* Destination is button, set the "click" at its assigned threshold */
+        if (config.in_cfg[dev].map_cfg[i].src_btn == PAD_RM &&
+                config.in_cfg[dev].map_cfg[i].perc_threshold > perc_threshold_r) {
+            perc_threshold_r = config.in_cfg[dev].map_cfg[i].perc_threshold;
+            map_cnt_r++;
+        }
+        else if (config.in_cfg[dev].map_cfg[i].src_btn == PAD_LM &&
+                config.in_cfg[dev].map_cfg[i].perc_threshold > perc_threshold_l) {
+            perc_threshold_l = config.in_cfg[dev].map_cfg[i].perc_threshold;
+            map_cnt_l++;
+        }
+    }
+    /* If only one mapping exist do not set resistance */
+    if (map_cnt_r < 2) {
```

Comment:
If we have only one mapping for that source, we do nothing.

Otherwise the trigger always ended with resistance set to 50% on all systems. I found that more annoying than helpful.

Reply:
Makes sense to me.

Reply:
Actually I think we need to redo the logic a bit.

We want to increment the count even if it's mapped to an analog axis, I think we want to increment it both when it's digital and analog.